src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.tool.TaskList;
+
+/**
+ * Represents a storage manager to handle file I/O.
+ */
+public class Storage {
+
+    /**
+     * String representation of path that stores user data
+     */
+    protected String dataPath;
+
+    /**
+     * Creates a storage to certain path.
+     *
+     * @param dataPath Data path (relative).
+     */
+    public Storage(String dataPath) {
+        this.dataPath = dataPath;
+    }
+
+    /**
+     * Loads the task list from stored file.
+     *
+     * @return An array list that represents the task list.
+     * @throws IOException An exception when the system cannot create the file.
+     */
+    public ArrayList<Task> load() throws IOException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        try {
+            File dataFile = new File(dataPath);
+            Scanner sc = new Scanner(dataFile);
+
+            //Read the file
+            while (sc.hasNextLine()) {
+                String nowTask = sc.nextLine();
+
+                String[] taskComponents = nowTask.split(" / ");
+
+                //Decode the file representation of list
+                switch (taskComponents[0]) {
+                case "T":
+                    taskList.add(new Todo(taskComponents[1].equals("1"), taskComponents[2]));
+                    break;
+                case "E":
+                    taskList.add(new Event(taskComponents[1].equals("1"), taskComponents[2],
+                             LocalDateTime.parse(taskComponents[3])));
```

Comment:
This may just be the formatting being off on my screen, but there may be a newline between the arguments to the `Event` constructor. If all appears right on your screen, my apologies!

Reply:
You are right, I should change it.